pkg/kademlia/queue_test.go

```diff
@@ -4,60 +4,41 @@
 package kademlia
 
 import (
-	"container/heap"
 	"math/big"
+	"strings"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
-
+	"storj.io/storj/pkg/node"
 	"storj.io/storj/pkg/pb"
 )
 
-func TestPriorityQueue(t *testing.T) {
-	cases := []struct {
-		target   *big.Int
-		nodes    map[string]*pb.Node
-		pq       PriorityQueue
-		expected []int
-	}{
-		{
-			target: func() *big.Int {
-				i, ok := new(big.Int).SetString("0001", 2)
-				assert.True(t, ok)
-				return i
-			}(),
-			nodes: map[string]*pb.Node{
-				"1001": &pb.Node{Id: "1001"},
-				"0100": &pb.Node{Id: "0100"},
-				"1100": &pb.Node{Id: "1100"},
-				"0010": &pb.Node{Id: "0010"},
-			},
-			pq:       make(PriorityQueue, 4),
-			expected: []int{3, 5, 8, 13},
-		},
+//IDFromBinStr turns a string like '110001' into a string like 'a'
+func BinStr(s string) string {
+	b := []byte(strings.Repeat("0", 8-len(s)%8) + s)
+	a := make([]byte, len(b)/8)
+	for i := 0; i < len(b); i++ {
+		a[i/8] |= ((b[i] - '0') << uint(7-i%8))
 	}
+	return string(a)
+}
 
-	for _, v := range cases {
-		i := 0
-		for id, value := range v.nodes {
-			bn, ok := new(big.Int).SetString(id, 2)
-			assert.True(t, ok)
-			v.pq[i] = &Item{
-				value:    value,
-				priority: new(big.Int).Xor(v.target, bn),
-				index:    i,
-			}
-			i++
-		}
-		heap.Init(&v.pq)
-
-		i = 0
-		for v.pq.Len() > 0 {
-			item := heap.Pop(&v.pq).(*Item)
-			assert.Equal(t, big.NewInt(int64(v.expected[i])), item.priority)
-			i++
-		}
+func TestXorQueue(t *testing.T) {
```

Comment:
can we test going over the limit ?

Reply:
added test, also added return of nil when queue is empty and test